Ginger/GingerCoreNET/Run/RunListenerLib/CenteralizedExecutionLogger/AccountReportEntitiesDataMapping.cs

```diff
@@ -477,136 +549,295 @@ public static Amdocs.Ginger.CoreNET.Execution.eRunStatus GetRunnerStatus(GingerE
+            accountReportRunner.ChildsExecutionStatistics.Activity = new ActivitiesStatistics();
+            accountReportRunner.ChildsExecutionStatistics.BusinessFlow = new BusinessFlowStatistics();
+            
+                accountReportRunner.ChildsExecutionStatistics.Action.StatusCount = new Dictionary<string, int>();
+                accountReportRunner.ChildsExecutionStatistics.Activity.StatusCount = new Dictionary<string, int>();
+                accountReportRunner.ChildsExecutionStatistics.BusinessFlow.StatusCount = new Dictionary<string, int>();
 
-            accountReportRunner.ChildPassedItemsCount = new List<AccountReport.Contracts.Helpers.DictObject>();
-            accountReportRunner.ChildPassedItemsCount.Add(new AccountReport.Contracts.Helpers.DictObject
-            { Key = Actvities, Value = ChildPassedItemsCountActivity });
+                var businessFlowStat = (from a in ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows.GroupBy(x => x.RunStatus)
+                                        select new KeyValuePair<string, int>(a.First().RunStatus.ToString(), a.Count())).ToList();
 
-            accountReportRunner.ChildExecutableItemsCount.Add(new AccountReport.Contracts.Helpers.DictObject
-            { Key = Actions, Value = ChildExecutableItemsCountAction });
+                foreach (var bstat in businessFlowStat)
+                {
+                    accountReportRunner.ChildsExecutionStatistics.BusinessFlow.StatusCount[bstat.Key] = bstat.Value;
+                }
+                accountReportRunner.ChildsExecutionStatistics.BusinessFlow.TotalExecuted = ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows.Count(bf => bf.RunStatus == eRunStatus.Failed || bf.RunStatus == eRunStatus.Passed || bf.RunStatus == eRunStatus.FailIgnored || bf.RunStatus == eRunStatus.Stopped || bf.RunStatus == eRunStatus.Completed);
+                accountReportRunner.ChildsExecutionStatistics.BusinessFlow.TotalPassed = ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows.Count(bf => bf.RunStatus == eRunStatus.Passed);
+                accountReportRunner.ChildsExecutionStatistics.BusinessFlow.TotalExecutable = ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows.Count(bf => bf.Active && (bf.RunStatus == eRunStatus.Passed || bf.RunStatus == eRunStatus.Failed || bf.RunStatus == eRunStatus.FailIgnored || bf.RunStatus == eRunStatus.Blocked));
+                        
 
-            accountReportRunner.ChildExecutedItemsCount.Add(new AccountReport.Contracts.Helpers.DictObject
-            { Key = Actions, Value = ChildExecutedItemsCountAction });
+            foreach (BusinessFlow businessFlow in ((GingerExecutionEngine)gingerRunner.Executor).BusinessFlows)
+            {
+                if (businessFlow.Active)
+                {                   
+                    var bfStat = accountReportStatistics.Where(x => x.EntityId == businessFlow.Guid).FirstOrDefault();
```

Comment:
Use FirstOrDefault directly instead of where

Reply:
Done.